syntax = "proto3";

service HostNetworkModule {

    rpc CheckInterfaceStatus(CheckInterfaceStatusRequest) returns (CheckInterfaceStatusResponse) {};
    rpc GetIfaceConnectionStats(GetIfaceStatsRequest) returns (GetIfaceStatsResponse) {};
    rpc SetIfaceDown(SetIfaceRequest) returns (SetIfaceResponse) {};
    rpc SetIfaceUp(SetIfaceRequest) returns (SetIfaceResponse) {};
}

message CheckInterfaceStatusRequest {
	string iface_name = 1;
}

message CheckInterfaceStatusResponse {
    int32 code = 1;
    bool status = 2;
}

message GetIfaceStatsRequest {
	string iface_name = 1;
}

message GetIfaceStatsResponse {
    int32 code = 1;
    string stats = 2;
}

message SetIfaceRequest {
	string iface_name = 1;
}

message SetIfaceResponse {
    int32 code = 1;
    bool success = 2;
}

